src/features/search/components/CalendarPicker.tsx

```diff
@@ -0,0 +1,86 @@
+import { t } from '@lingui/macro'
+import DateTimePicker from '@react-native-community/datetimepicker'
+import React, { useState } from 'react'
+import { Platform } from 'react-native'
+import styled from 'styled-components/native'
+
+import { DateFilterModal } from 'features/offer/components/DateFilterModal'
+import { _ } from 'libs/i18n'
+import { formatToCompleteFrenchDate } from 'libs/parsers/formatDates'
+import { useModal } from 'ui/components/modals/useModal'
+import { ColorsEnum, getSpacing, Typo } from 'ui/theme'
+
+import { DateFilterButton } from '../atoms/DateFilterButton'
+
+export const CalendarPicker: React.FC = () => {
+  const [date, setDate] = useState(new Date(Date.now()))
+  const [mode, setMode] = useState<'time' | 'date' | undefined>('date')
+  const [show, setShow] = useState<boolean>(false)
+  const [isSelected, setIsSelected] = useState<boolean>(false)
+
+  const onChange = (event: Event, selectedDate: Date | undefined) => {
+    const chosenDate = selectedDate || date
+    setShow(Platform.OS === 'ios')
+    setDate(chosenDate)
+  }
+
+  const showMode = (currentMode: 'time' | 'date' | undefined) => {
+    setShow(true)
+    setMode(currentMode)
+    setIsSelected(true)
+  }
+
+  const showDatepicker = () => {
+    showMode('date')
+    showDateFilterModal()
+  }
+
+  const { visible: dateFilterModalVisible, showModal: showDateFilterModal, hideModal } = useModal(
```

Comment:
pas grand chose mais:
```suggestion
  const { visible, showModal: showDateFilterModal, hideModal } = useModal(false)
```

et à mettre avec les autres hooks

Reply:
si on retire le préfixe sur 'visible' et 'hidemodal' autant le faire sur 'showmodal' par souci de cohérence